app/uk/gov/hmrc/apinotificationqueue/controller/QueueController.scala

```diff
@@ -25,55 +25,67 @@ import play.api.http.HttpEntity
 import play.api.libs.json.Json
 import play.api.mvc._
 import uk.gov.hmrc.apinotificationqueue.model.{Notification, Notifications}
-import uk.gov.hmrc.apinotificationqueue.service.QueueService
-import uk.gov.hmrc.http.BadRequestException
+import uk.gov.hmrc.apinotificationqueue.service.{ApiSubscriptionFieldsService, QueueService}
+import uk.gov.hmrc.http.HeaderCarrier
 import uk.gov.hmrc.play.bootstrap.controller.BaseController
 import uk.gov.hmrc.play.http.logging.MdcLoggingExecutionContext._
 
 import scala.concurrent.Future
 
 @Singleton()
-class QueueController @Inject()(queueService: QueueService, idGenerator: NotificationIdGenerator) extends BaseController {
+class QueueController @Inject()(queueService: QueueService, fieldsService: ApiSubscriptionFieldsService, idGenerator: NotificationIdGenerator) extends BaseController {
 
-  private val CLIENT_ID_HEADER_NAME = "X-Client-ID"
-  private val CLIENT_ID_REQUIRED_ERROR = new BadRequestException(s"$CLIENT_ID_HEADER_NAME required.")
+  val SUBSCRIPTION_FIELD_HEADER_NAME = "api-subscription-fields-id"
```

Comment:
why not making these 2 val as `private`?
I understood you prefer to use local constants